lib/pbench/test/unit/agent/task/test_results_push.py

```diff
@@ -23,13 +28,36 @@ class TestResultsPush:
     URL = "http://pbench.example.com/api/v1"
 
     @staticmethod
-    def add_http_mock_response(code: HTTPStatus = HTTPStatus.OK, message: str = ""):
-        responses.add(
-            responses.PUT,
-            f"{TestResultsPush.URL}/upload/{os.path.basename(tarball)}",
-            status=code,
-            json={"message": message},
-        )
+    def add_http_mock_response(
+        status_code: HTTPStatus = None, message: Union[str, Dict, type(None)] = None
+    ):
+        if status_code:
+            if message is None:
+                responses.add(
+                    responses.PUT,
+                    f"{TestResultsPush.URL}/upload/{os.path.basename(tarball)}",
+                    status=status_code,
+                )
+            elif isinstance(message, dict):
+                responses.add(
+                    responses.PUT,
+                    f"{TestResultsPush.URL}/upload/{os.path.basename(tarball)}",
+                    status=status_code,
+                    json=message,
+                )
+            # I can only add a `json' attribute to the response. Trying `text' or `reason' makes
+            # responses.add() barf:
```

Comment:
For a text body, I think you want the [`body` keyword](https://github.com/getsentry/responses/blob/master/README.rst#response-parameters) instead of `json`.

Reply:
OK - I'm not going to worry about these tests in this PR. I'm more likely to break other things at this point (see below). I'll leave them for the next go-around.

Reply:
See [an option below](https://github.com/distributed-system-analysis/pbench/pull/3348#discussion_r1160994703).

Reply:
Yes, `body` seems to do the trick. I did add some string cases to `test_results_push.py`. There is still a lot that could be done though.